### ISSUE TITLE
[Broken] stream.nbcolympics.com

### ISSUE THREAD

**Issue by frijole** [yt-dlp/yt-dlp#989]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running yt-dlp version **2021.09.02**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
[debug] Command-line config: ['-v', '--ap-mso=Comcast_SSO', '--ap-username', 'PRIVATE', '--ap-password', 'PRIVATE', '--cookies', 'cookies.txt', '-F', 'https://stream.nbcolympics.com/artistic-swimming-team-free-routine', '--ignore-errors']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] yt-dlp version 2021.09.02 
[debug] Python version 3.9.7 (CPython 64bit) - macOS-10.15.7-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.4, ffprobe 4.4, phantomjs 2.1.1, rtmpdump 2.4
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
[debug] [nbcolympics:stream] Extracting URL: https://stream.nbcolympics.com/artistic-swimming-team-free-routine
[nbcolympics:stream] artistic-swimming-team-free-routine: Downloading webpage
[nbcolympics:stream] 2022506: Downloading event config
[nbcolympics:stream] 2022506: Downloading leap config
[nbcolympics:stream] 2022506: Downloading Provider Redirect Page
[nbcolympics:stream] 2022506: Logging in
[nbcolympics:stream] 2022506: Retrieving Session
ERROR: [nbcolympics:stream] artistic-swimming-team-free-routine: Unable to download webpage: HTTP Error 401: Unauthorized (caused by <HTTPError 401: 'Unauthorized'>); please report this issue on  https://github.com/yt-dlp/yt-dlp . Make sure you are using the latest version; see  https://github.com/yt-dlp/yt-dlp  on how to update. Be sure to call yt-dlp with the --verbose flag and include its complete output.
  File "/usr/local/Cellar/yt-dlp/2021.9.2/libexec/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 693, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/local/Cellar/yt-dlp/2021.9.2/libexec/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 3218, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 561, in error
    return self._call_chain(*args)
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/urllib/request.py", line 641, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
```

## Description

Trying to download videos or even list formats available from `stream.nbcolympics.com` such as https://stream.nbcolympics.com/artistic-swimming-team-free-routine or https://stream.nbcolympics.com/peacock-210724-tokyo-tonight fails with `Error 401: Unauthorized` even with a valid cable credentials and cookies.txt from a live browser session playing the same video.

**Comment by pukkandan** [comment]
try without cookies. It's unlikely that's the issue, but it may sometimes help

**Comment by pukkandan** [comment]
@wesnm may be interested since u have worked on this extractor before

**Comment by wesnm** [comment]
This URL downloads fine for me with my sling tv login. It may be something in the Comcast MSO code independent of the NBC extractor. Can you try some other site that uses your MSO?

```[nbcolympics:stream] 2022506: Retrieving Session
[nbcolympics:stream] 2022506: Retrieving Authorization Token
[nbcolympics:stream] 2022506: Retrieving Media Token
[nbcolympics:stream] 2022506: Retrieving tokenized URL
[nbcolympics:stream] 2022506: Downloading m3u8 information
[info] 2022506: Downloading 1 format(s): 6596-1
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 1162
[download] Destination: Team - Free Routine (Final) [2022506].mp4
[download]   0.2% of ~5.32GiB at  1.93MiB/s ETA 01:13:51^C
ERROR: Interrupted by user
```

**Comment by llacb47** [comment]
works for me (comcast is my ISP)

```
yt-dlp --ignore-config -vF --ap-mso=Comcast_SSO https://stream.nbcolympics.com/artistic-swimming-team-free-routine --proxy "http://localhost:8888"
[debug] Command-line config: ['--ignore-config', '-vF', '--ap-mso=Comcast_SSO', 'https://stream.nbcolympics.com/artistic-swimming-team-free-routine', '--proxy', 'http://localhost:8888']
[debug] Encodings: locale cp1252, fs utf-8, out utf-8, pref cp1252
[debug] yt-dlp version 2021.09.02
[debug] Python version 3.9.6 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] exe versions: ffmpeg n4.3.2-163-g6c414cf8f7, ffprobe n4.3.2-163-g6c414cf8f7
[debug] Optional libraries: mutagen, pycryptodome, sqlite, websockets
[debug] Proxy map: {'http': 'http://localhost:8888', 'https': 'http://localhost:8888'}
[debug] [nbcolympics:stream] Extracting URL: https://stream.nbcolympics.com/artistic-swimming-team-free-routine
[nbcolympics:stream] artistic-swimming-team-free-routine: Downloading webpage
[nbcolympics:stream] 2022506: Downloading event config
[nbcolympics:stream] 2022506: Downloading leap config
[nbcolympics:stream] 2022506: Retrieving Media Token
[nbcolympics:stream] 2022506: Retrieving tokenized URL
[nbcolympics:stream] 2022506: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 2022506:
ID     EXT RESOLUTION FPS |   TBR PROTO  | VCODEC        VBR ACODEC     ABR
------ --- ---------- --- - ----- ------ - ----------- ----- --------- ----
221-0  mp4 256x144    30  |  221k m3u8_n | avc1.4d400c  221k mp4a.40.2   0k
221-1  mp4 256x144    30  |  221k m3u8_n | avc1.4d400c  221k mp4a.40.2   0k
346-0  mp4 384x216    30  |  346k m3u8_n | avc1.4d400d  346k mp4a.40.2   0k
346-1  mp4 384x216    30  |  346k m3u8_n | avc1.4d400d  346k mp4a.40.2   0k
596-0  mp4 512x288    30  |  596k m3u8_n | avc1.4d4015  596k mp4a.40.2   0k
596-1  mp4 512x288    30  |  596k m3u8_n | avc1.4d4015  596k mp4a.40.2   0k
996-0  mp4 640x360    30  |  996k m3u8_n | avc1.4d401e  996k mp4a.40.2   0k
996-1  mp4 640x360    30  |  996k m3u8_n | avc1.4d401e  996k mp4a.40.2   0k
1896-0 mp4 896x504    30  | 1896k m3u8_n | avc1.4d401f 1896k mp4a.40.2   0k
1896-1 mp4 896x504    30  | 1896k m3u8_n | avc1.4d401f 1896k mp4a.40.2   0k
3096-0 mp4 1024x576   30  | 3096k m3u8_n | avc1.64001f 3096k mp4a.40.2   0k
3096-1 mp4 1024x576   30  | 3096k m3u8_n | avc1.64001f 3096k mp4a.40.2   0k
4596-0 mp4 1280x720   30  | 4596k m3u8_n | avc1.64001f 4596k mp4a.40.2   0k
4596-1 mp4 1280x720   30  | 4596k m3u8_n | avc1.64001f 4596k mp4a.40.2   0k
6596-0 mp4 1920x1080  30  | 6596k m3u8_n | avc1.640028 6596k mp4a.40.2   0k
6596-1 mp4 1920x1080  30  | 6596k m3u8_n | avc1.640028 6596k mp4a.40.2   0k
```

**Comment by pukkandan** [comment]
Closed due to inactivity
